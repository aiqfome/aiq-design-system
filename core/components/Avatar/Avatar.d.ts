import React from 'react';
import { Props as BoxProps } from '../Box';
export interface Props extends BoxProps {
    src?: string;
    palette?: string;
    alt?: string;
    size?: string;
    fallback?: string;
    variant?: 'box' | 'rounded';
}
export declare const AvatarStyled: import("styled-components").StyledComponent<"img", any, {}, never>;
export declare const Avatar: React.FC<Props>;
//# sourceMappingURL=Avatar.d.ts.map