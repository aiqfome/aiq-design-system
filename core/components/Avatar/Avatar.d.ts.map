{"version":3,"file":"Avatar.d.ts","sourceRoot":"","sources":["../../../src/components/Avatar/Avatar.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAsC,MAAM,OAAO,CAAA;AAM1D,OAAO,EAAO,KAAK,IAAI,QAAQ,EAAE,MAAM,QAAQ,CAAA;AAE/C,MAAM,WAAW,KAAM,SAAQ,QAAQ;IACrC,GAAG,CAAC,EAAE,MAAM,CAAA;IACZ,OAAO,CAAC,EAAE,MAAM,CAAA;IAChB,GAAG,CAAC,EAAE,MAAM,CAAA;IACZ,IAAI,CAAC,EAAE,MAAM,CAAA;IACb,QAAQ,CAAC,EAAE,MAAM,CAAA;IACjB,OAAO,CAAC,EAAE,KAAK,GAAG,SAAS,CAAA;CAC5B;AAaD,eAAO,MAAM,YAAY,oEAGxB,CAAA;AAUD,eAAO,MAAM,MAAM,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAqDlC,CAAA"}