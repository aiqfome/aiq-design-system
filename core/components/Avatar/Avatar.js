"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Avatar = exports.AvatarStyled = void 0;
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importDefault(require("styled-components"));
const Text_1 = require("../Text");
const Box_1 = require("../Box");
const BoxStyled = styled_components_1.default(Box_1.Box) `
  & > img,
  & {
    border-radius: ${({ variant }) => (variant === 'rounded' ? '50%' : '5px')};
    width: ${({ size }) => size || '36px'};
    height: ${({ size }) => size || '36px'};
    object-fit: ${({ variant }) => variant === 'rounded' ? 'cover' : 'contain'};
  }
`;
exports.AvatarStyled = styled_components_1.default.img `
  width: 36px;
  height: 36px;
`;
const AvatarInitial = react_1.memo(({ alt = '', palette }) => {
    return (react_1.default.createElement(Text_1.Text, { "data-testid": 'name', fontWeight: 'bold', color: palette }, alt.substring(0, 1).toUpperCase()));
});
exports.Avatar = (_a) => {
    var { src, alt, fallback, variant = 'box', palette = 'primary' } = _a, props = __rest(_a, ["src", "alt", "fallback", "variant", "palette"]);
    const [useFallback, setUseFallback] = react_1.useState(false);
    const [imgFallback, setImgFallback] = react_1.useState(fallback);
    const handleSrcError = react_1.useCallback(() => setUseFallback(true), []);
    const getImage = () => {
        if (src && !useFallback) {
            return (react_1.default.createElement(exports.AvatarStyled, { src: src, alt: alt, "data-testid": 'src', onError: handleSrcError }));
        }
        if (imgFallback) {
            return (react_1.default.createElement(exports.AvatarStyled, { alt: alt, src: imgFallback, "data-testid": 'src', onError: () => setImgFallback('') }));
        }
        return react_1.default.createElement(AvatarInitial, { alt: alt, palette: palette });
    };
    return (react_1.default.createElement(BoxStyled, Object.assign({ display: 'flex', variant: variant, "data-testid": 'box', alignItems: 'center', verticalAlign: 'center', justifyContent: 'center', backgroundColor: src && !useFallback ? 'transparent' : `${palette}Light` }, props), getImage()));
};
exports.Avatar.propTypes = {
    src: prop_types_1.default.string,
    fallback: prop_types_1.default.string,
    palette: prop_types_1.default.string,
    alt: prop_types_1.default.string.isRequired,
    variant: prop_types_1.default.oneOf(['box', 'rounded'])
};
exports.Avatar.defaultProps = {
    palette: 'primary'
};
AvatarInitial.displayName = 'AvatarInitial';
AvatarInitial.propTypes = {
    alt: prop_types_1.default.string.isRequired,
    palette: prop_types_1.default.string
};
