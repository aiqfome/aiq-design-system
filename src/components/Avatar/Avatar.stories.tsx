import React from 'react'
import { withKnobs, text, select } from '@storybook/addon-knobs'

import { Avatar } from './Avatar'

export default {
  component: Avatar,
  title: 'Avatar',
  decorators: [withKnobs]
}

export const Basic: React.FC = () => (
  <Avatar
    alt={text('Label', 'Avatar')}
    src={text('src', '')}
    palette={select(
      'palette',
      { Primary: 'primary', Secondary: 'secondary' },
      'primary'
    )}
  />
)

export const WithUrl: React.FC = () => (
  <Avatar
    palette={select(
      'palette',
      { Primary: 'primary', Secondary: 'secondary' },
      'primary'
    )}
    src={text(
      'src',
      'https://lh3.googleusercontent.com/rALGk_PU3JMf_5NS5FEYScz9zxgjRBNePvMheCnHIO_lrSs089QcwguwqRVaDLWWAQ'
    )}
    alt={text('alt', 'Avatar')}
  />
)

export const Rounded: React.FC = () => (
  <Avatar
    palette={select(
      'palette',
      { Primary: 'primary', Secondary: 'secondary' },
      'primary'
    )}
    alt={text('alt', 'Avatar')}
    variant='rounded'
  />
)
