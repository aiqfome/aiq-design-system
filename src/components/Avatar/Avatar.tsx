import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Text } from '../Text'
import { Box, Props as BoxProps } from '../Box'

export interface Props extends BoxProps {
  src?: string
  palette?: string
  alt: string
  variant?: 'box' | 'rounded'
}

export const AvatarStyled = styled.img`
  width: 36px;
  height: 36px;
`

export const Avatar: React.FC<Props> = ({
  src,
  alt,
  palette = 'primary',
  variant = 'box',
  ...props
}) => {
  const [name, setName] = useState('')

  useEffect(() => {
    setName(alt.substring(0, 1).toUpperCase())
  }, [alt])

  return (
    <Box
      width={variant === 'box' ? '36px' : '32px'}
      height={variant === 'box' ? '36px' : '32px'}
      display='flex'
      verticalAlign='center'
      alignItems='center'
      justifyContent='center'
      borderRadius={variant === 'box' ? '5px' : '32px'}
      backgroundColor={src ? 'transparent' : `${palette}Light`}
      {...props}
    >
      {src ? (
        <AvatarStyled src={src} alt={alt} />
      ) : (
        <Text fontWeight='bold' color={palette}>
          {name}
        </Text>
      )}
    </Box>
  )
}

Avatar.propTypes = {
  src: PropTypes.string,
  palette: PropTypes.string,
  alt: PropTypes.string.isRequired,
  variant: PropTypes.oneOf(['box', 'rounded'])
}
