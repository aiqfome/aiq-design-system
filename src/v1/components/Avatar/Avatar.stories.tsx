import React from 'react'

import { text, select } from '@storybook/addon-knobs'

import { Avatar } from './Avatar'
import { createPageExport } from '@/src/utils/storybook'

const aiqProps = [
  'src',
  'alt',
  'children',
  'palette',
  'size',
  'fallback',
  'variant'
]

export default createPageExport(Avatar, 'Avatar v1', aiqProps, {
  argTypes: {
    variant: {
      control: 'select',
      options: ['box', 'rounded']
    }
  },
  args: {
    alt: 'Avatar',
    size: '32px',
    children: 'Avatar',
    palette: 'primary'
  }
})

export const Basic = props => (
  <Avatar
    alt={text('Label', 'Avatar')}
    src={text('src', '')}
    palette={select(
      'palette',
      { Primary: 'primary', Secondary: 'secondary' },
      'primary'
    )}
    children={text('children', 'Avatar')}
    {...props}
  />
)
Basic.args = {
  variant: 'box'
}

export const WithUrl = args => (
  <Avatar
    palette={select(
      'palette',
      { Primary: 'primary', Secondary: 'secondary' },
      'primary'
    )}
    src={text(
      'src',
      'https://lh3.googleusercontent.com/rALGk_PU3JMf_5NS5FEYScz9zxgjRBNePvMheCnHIO_lrSs089QcwguwqRVaDLWWAQ'
    )}
    alt={text('alt', 'Avatar')}
    children={text('children', 'Avatar')}
    {...args}
  />
)
WithUrl.args = {
  src: 'https://lh3.googleusercontent.com/rALGk_PU3JMf_5NS5FEYScz9zxgjRBNePvMheCnHIO_lrSs089QcwguwqRVaDLWWAQ',
  variant: 'box'
}

export const Rounded = args => (
  <Avatar
    palette={select(
      'palette',
      { Primary: 'primary', Secondary: 'secondary' },
      'primary'
    )}
    alt={text('alt', 'Avatar')}
    variant='rounded'
    children={text('children', 'Avatar')}
    {...args}
  />
)
Rounded.args = {
  variant: 'rounded'
}
