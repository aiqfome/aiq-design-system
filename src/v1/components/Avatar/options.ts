const testId = {
  root: 'avatar-root',
  image: 'avatar-img',
  text: 'avatar-name'
}

export { testId }