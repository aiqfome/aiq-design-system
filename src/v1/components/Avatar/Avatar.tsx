import React, { useCallback, useState, memo, PropsWithChildren } from 'react'

import { stitches } from '@/src/providers'

import { Text } from '@/src/components/Text'
import { Box, Props as BoxProps } from '@/src/components/Box'
import { testId } from './options'

const { css, styled } = stitches

const BoxRootContainer = stitches.styled(Box, {})

export type Props = BoxProps &
  PropsWithChildren<{
    src?: string
    palette?: string
    alt?: string
    size?: string
    fallback?: string
    variant?: 'box' | 'rounded'
  }>

function BoxRoot({ variant, size, ...props }: Props) {
  const isRounded = variant === 'rounded'

  const className = css({
    br: isRounded ? '50%' : '5px',
    size: size || '36px',
    objectFit: isRounded ? 'cover' : 'contain'
  })()

  return <BoxRootContainer className={className} {...props} />
}

export const AvatarStyled = styled('img', {
  width: '36px',
  height: '36px'
})

const AvatarInitial: React.FC<Props> = memo(({ alt, palette, children }) => {
  return (
    <Text
      data-testid={testId.text}
      aria-label={alt}
      fontWeight='bold'
      color={palette}
    >
      {children.substring(0, 1).toUpperCase()}
    </Text>
  )
})

const AvatarImage = stitches.styled('img', {
  width: '36px',
  height: '36px'
})

const Avatar: React.FC<Props> = ({
  src,
  alt,
  fallback,
  variant = 'box',
  palette = 'primary',
  children,
  ...props
}) => {
  const [useFallback, setUseFallback] = useState(false)
  const [imgFallback, setImgFallback] = useState(fallback)

  const handleSrcError = useCallback(() => setUseFallback(true), [])

  const Image = props => <AvatarImage data-testid={testId.image} {...props} />

  const getImage = () => {
    if (src && !useFallback) {
      return <Image src={src} alt={alt} onError={handleSrcError} />
    }

    if (imgFallback) {
      return (
        <Image alt={alt} src={imgFallback} onError={() => setImgFallback('')} />
      )
    }

    return <AvatarInitial palette={palette}>{children}</AvatarInitial>
  }

  return (
    <BoxRoot
      display='flex'
      variant={variant}
      data-testid={testId.root}
      alignItems='center'
      verticalAlign='center'
      justifyContent='center'
      backgroundColor={src && !useFallback ? 'transparent' : `${palette}Light`}
      {...props}
    >
      {getImage()}
    </BoxRoot>
  )
}

export { Avatar }

Avatar.defaultProps = {
  palette: 'primary'
}

AvatarInitial.displayName = 'AvatarInitial'
