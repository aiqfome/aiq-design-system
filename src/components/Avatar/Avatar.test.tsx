import React from 'react'
import renderer from 'react-test-renderer'

import { Avatar } from './Avatar'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Avatar alt='avatar component!' />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
