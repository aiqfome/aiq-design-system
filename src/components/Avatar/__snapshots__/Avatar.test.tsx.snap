// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-AxirZ eGmxmb"
  display="flex"
  height="36px"
  width="36px"
>
  <span
    className="sc-AxjAm gdbfbt"
    color="primary"
    fontWeight="bold"
  >
    
  </span>
</div>
`;
